### PR TITLE
Refactor Folders

### PR DESCRIPTION
**Pull request by pablogs9**
This PR restructures the folders in order to have one advanced tutorials per RTOS

### CODE REVIEW

**Comment by pablogs9** [comment]
@ralph-lange it is OK this organization for you?
What are we going to do with the `tutorials/basic` folder?

File: _data/docs.yml [hunk 1]
```diff
@@ -39,20 +39,30 @@
...
-   - tutorials/basic/getting_started
+   - tutorials/core/getting_started
    - tutorials/basic/getting_started_embedded
    - tutorials/basic/getting_started_simulator
-   - tutorials/basic/microxrcedds_rmw_configuration
+   - tutorials/core/programming_rcl_rclc
+   - tutorials/core/microxrcedds_rmw_configuration
 
-- title: Advanced Tutorials
+- title: Advanced Tutorials With Nuttx
```
Comment (ralph-lange) [inline _data/docs.yml]:
```suggestion
- title: Advanced Tutorials with Nuttx
```

File: _data/docs.yml [hunk 2]
```diff
@@ -39,20 +39,30 @@
...
-   - tutorials/advanced/microros_6lowpan_tutorial
+   - tutorials/advanced/nuttx/nuttx_getting_started
+   - tutorials/advanced/nuttx/debugging_gdb_openocd
+   - tutorials/advanced/nuttx/debugging_vscode
+   - tutorials/advanced/nuttx/6lowpan_nuttx_rpi
+   - tutorials/advanced/nuttx/tracing
+   - tutorials/advanced/nuttx/microros_nuttx_bsp
+   - tutorials/advanced/nuttx/microros_6lowpan_tutorial
+
+- title: Advanced Tutorials With FreeRTOS
```
Comment (ralph-lange) [inline _data/docs.yml]:
```suggestion
- title: Advanced Tutorials with FreeRTOS
```

File: _data/docs.yml [hunk 3]
```diff
@@ -39,20 +39,30 @@
...
+  docs:
+    - tutorials/advanced/freertos/freertos_getting_started
+
+- title: Advanced Tutorials With Zephyr
```
Comment (ralph-lange) [inline _data/docs.yml]:
```suggestion
- title: Advanced Tutorials with Zephyr
```

File: _data/docs.yml [hunk 4]
```diff
@@ -39,20 +39,30 @@
...
+  docs:
+    - tutorials/advanced/zephyr/zephyr_getting_started
+  
 
 - title: Demos
```
Comment (ralph-lange) [inline _data/docs.yml]: I propose to move the demos before the advanced tutorials, i.e. directly after the core tutorials.

File: _data/docs.yml [hunk 5]
```diff
@@ -39,20 +39,30 @@
 
 - title: Basic Tutorials
   docs:
-   - tutorials/basic/getting_started
+   - tutorials/core/getting_started
    - tutorials/basic/getting_started_embedded
```
Comment (ralph-lange) [inline _data/docs.yml]:
```suggestion
   - tutorials/core/getting_started_embedded
```
We should rename the folder `basic` completely to `core`.

File: _data/docs.yml [hunk 6]
```diff
@@ -39,20 +39,30 @@
...
-   - tutorials/advanced/nuttx_getting_started
-   - tutorials/advanced/debugging_gdb_openocd
-   - tutorials/advanced/debugging_vscode
-   - tutorials/advanced/6lowpan_nuttx_rpi
-   - tutorials/advanced/tracing
-   - tutorials/advanced/microros_nuttx_bsp
-   - tutorials/advanced/microros_6lowpan_tutorial
+   - tutorials/advanced/nuttx/nuttx_getting_started
+   - tutorials/advanced/nuttx/debugging_gdb_openocd
+   - tutorials/advanced/nuttx/debugging_vscode
```
Comment (ralph-lange) [inline _data/docs.yml]: Nitpicking: I'm unsure whether `advanced/nuttx/` or `advanced_nuttx/` is better? The first option looks nicer, the second option reflects the menu nesting better. If you decide for the first option, please ensure that https://micro-ros.github.io/docs/tutorials/advanced/ is somehow redirected and does not return an HTTP 404.

File: _data/docs.yml (hunk 5 above)
Comment (pablogs9) [reply _data/docs.yml]: Done

File: _data/docs.yml (hunk 6 above)
Comment (pablogs9) [reply _data/docs.yml]: Done

File: _data/docs.yml (hunk 4 above)
Comment (pablogs9) [reply _data/docs.yml]: Done

**Comment by pablogs9** [comment]
@ralph-lange it can't figure out why is the CI failing... can you see something?

**Comment by ralph-lange** [comment]
Hi @pablogs9, note that the segfault occurred after htmlproof has completed successfully. Therefore, I simply re-triggered the CI and this time no segfault occurred.

**Comment by pablogs9** [comment]
Nice, thanks! Squash merge it when you are ok with the changes.

**Comment by christophebedard** [comment]
May I ask why the "Getting started with ROS 2 tracing" tutorial was moved under "Advanced Tutorials with NuttX"?

Also, it's unfortunate that the link we put out there for the tutorial (https://micro-ros.github.io/docs/tutorials/advanced/tracing/) now returns a 404.

**Comment by pablogs9** [comment]
Please open a PR with your required modifications, I think that it is possible to redirect the URL you refer to the actual URL in the case we decide to keep it under Advanced tutorials.

What is your opinion @ralph-lange?

**Comment by christophebedard** [comment]
Sure, I can create the redirection page.

Could we create another "Advanced Tutorials" section, e.g. "Other Advanced Tutorials"?

**Comment by ralph-lange** [comment]
I'll fix this immediately. I should be moved into a separate menu "Advanced Tutorials with Linux". I'll also create a redirect.

**Pull request merged**